packages/jazz-tools/src/tools/coValues/coVector.ts

```diff
@@ -0,0 +1,696 @@
+
+    if (
+      this.requiredDimensionsCount !== undefined &&
+      vector.length !== this.requiredDimensionsCount
+    ) {
+      throw new Error(
+        `Vector dimension mismatch! CoVector '${this.$jazz.raw.id}' loaded with ${vector.length} dimensions, but the schema requires ${this.requiredDimensionsCount} dimensions`,
+      );
+    }
+
+    this._loadedVector = vector;
+
+    return vector;
+  }
+
+  /**
+   * Get a JSON representation of the `CoVector`
+   * @category Content
+   */
+  toJSON(): Array<number> {
+    return Array.from(this.vector);
+  }
+
+  valueOf() {
+    return this.vector as this;
+  }
+
+  /** @internal */
+  [inspect]() {
+    return this.toJSON();
+  }
+
+  [Symbol.toPrimitive]() {
+    return this.vector;
+  }
+
+  /**
+   * Load a `CoVector`
+   *
+   * @category Subscription & Loading
+   * @deprecated Use `co.vector(...).load` instead.
+   */
+  static async load<C extends CoVector>(
+    this: CoValueClass<C>,
+    id: ID<C>,
+    options?: {
+      loadAs?: Account | AnonymousJazzAgent;
+    },
+  ): Promise<CoVector | null> {
+    const coVector = await loadCoValueWithoutMe(this, id, options);
+
+    /**
+     * We are only interested in the entire vector. Since most vectors are small (<15kB),
+     * we can wait for the stream to be complete before returning the vector
+     */
+    if (!coVector?.$jazz.raw.isBinaryStreamEnded()) {
+      return new Promise<CoVector>((resolve) => {
+        subscribeToCoValueWithoutMe(
+          this,
+          id,
+          options || {},
+          (value, unsubscribe) => {
+            if (value.$jazz.raw.isBinaryStreamEnded()) {
+              unsubscribe();
+              resolve(value);
+            }
+          },
+        );
+      });
+    }
+
+    return coVector;
+  }
+
+  /**
+   * Subscribe to a `CoVector`, when you have an ID but don't have a `CoVector` instance yet
+   * @category Subscription & Loading
+   * @deprecated Use `co.vector(...).subscribe` instead.
+   */
+  static subscribe<V extends CoVector, const R extends RefsToResolve<V>>(
+    this: CoValueClass<V>,
+    id: ID<V>,
+    listener: (value: Resolved<V, R>, unsubscribe: () => void) => void,
+  ): () => void;
+  static subscribe<V extends CoVector, const R extends RefsToResolve<V>>(
+    this: CoValueClass<V>,
+    id: ID<V>,
+    options: SubscribeListenerOptions<V, R>,
+    listener: (value: Resolved<V, R>, unsubscribe: () => void) => void,
+  ): () => void;
+  static subscribe<V extends CoVector, const R extends RefsToResolve<V>>(
+    this: CoValueClass<V>,
+    id: ID<V>,
+    ...args: SubscribeRestArgs<V, R>
+  ): () => void {
+    const { options, listener } = parseSubscribeRestArgs(args);
+    return subscribeToCoValueWithoutMe<V, R>(this, id, options, listener);
+  }
+
+  // Vector operations
+  /**
+   * Calculate the magnitude of a vector.
+   */
+  static magnitude(vector: Float32Array | CoVector): number {
+    return Math.sqrt(vector.reduce((s, x) => s + x * x, 0));
+  }
```

Comment:
Any reason to provide both static and instance methods? Except for loading/subscriptions, CoValue classes implement all operations as instance methods.

Reply:
I assume the instance methods will be the most used (as they're easily available).

The static ones are present because I wanted the calculation functions to:
1. be pure with explicit inputs
2. be available as vector utility functions beyond CoVector instances (e.g. _give me magnitude of this random vector which doesn't need to be synced so it's not a CoVector instance_)

Does that go against the design principles here? I admit the (2) might be over-assuming use cases though